model.py

```diff
@@ -6069,14 +6069,12 @@ def get(cls, _db, url, do_get=None, extra_request_headers=None,
             media_type = None
 
         # At this point we can create/fetch a Representation object if
-        # we don't have one already, or if the URL or media type we
-        # actually got from the server differs from what we thought
-        # we had.
+        # we don't have one already, or if the URL we actually got from 
+        # the server differs from what we thought we had.
         if (not usable_representation 
-            or media_type != representation.media_type
             or url != representation.url):
             representation, is_new = get_one_or_create(
-                _db, Representation, url=url, media_type=media_type)
+                _db, Representation, url=url)
```

Comment:
I think it would be better to pass in a `presumed_media_type` to `get()` to handle the case where the server doesn't specify a media type but we know ahead of time what it probably is. That way we don't have a situation where a `Resource` is (temporarily) without a media type.

The url+media type constraint is designed to handle a case where a data source distinguishes between formats using content negotiation rather than distinctive URLs. So far this hasn't been an issue but it's theoretically possible and I wanted to account for it in core.